### ISSUE TITLE
Failed autocorrection for Style/StringLiteralsInInterpolation if string is non-ASCII

### ISSUE THREAD

**Issue by yaxartes** [rubocop-hq/rubocop#5475]
`rubocop --auto-correct` is supposed to change double quotes in interpolation
```
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements["ФИО лида ФЛ"]}']"
```
to single quotes
```
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements['ФИО лида ФЛ']}']"
```

Istead it leaves double quotes and changes non-ASCII characters (cyrillic in my case) to hex code
```
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements["\xD0\xA4\xD0\x98\xD0\x9E \xD0\xBB\xD0\xB8\xD0\xB4\xD0\xB0 \xD0\xA4\xD0\x9B"]}']"
```

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.3 x64-mingw32)
```

**Comment by asherkach** [comment]
Unfortunately, I can't reproduce this failure, @yaxartes. Is the character conversion from cyrillic to hex code the result of another cop performing an auto-correct?
```
$ cat test.rb
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements["ФИО лида ФЛ"]}']"
$ bundle exec rubocop --auto-correct --only Style/StringLiteralsInInterpolation test.rb
Inspecting 1 file
C

Offenses:

test.rb:1:68: C: [Corrected] Style/StringLiteralsInInterpolation: Prefer single-quoted strings inside interpolations.
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements["ФИО лида ФЛ"]}']"
                                                                   ^^^^^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
$ cat test.rb
@elements["Заголовок с ФИО лида ФЛ"] = "//span[text()='#{@elements['ФИО лида ФЛ']}']"
```

**Comment by yaxartes** [comment]
@asherkach I'm not 100% sure, but this was the warning
```
Style/IfUnlessModifier: Favor modifier `if` usage when having a single-line body. Another good alternative is the usage of control flow `&&`/`||`. (convention:Style/IfUnlessModifier)
```
And autocorrection doesn't change this line after I've added this to my .rubocop.yml config
```
Style/StringLiteralsInInterpolation: 
  Description: 'Enforces single quotes in string interpolation'
  Enabled: false
```

**Comment by pocke** [comment]
@yaxartes How do you execute RuboCop? From shell, editor or IDE? Please tell me the command if it is possible.
And I guess it is an encoding problem. So can you execute the following commands and tell me the outputs?

```bash
$ echo $LANG
$ ruby -e 'p Encoding.default_external'
```

I cannot reproduce it also. And I tried reproducing it in `LANG=C` environment, but I cannot.

**Comment by yaxartes** [comment]
@pocke Sorry If I wasn't clear enough. I run folowing command in shell in my project's directory:
```
rubocop --auto-correct
```

Output of your commands:
```
C:\WORK\tests>echo $LANG
$LANG

C:\WORK\tests>ruby -e 'p Encoding.default_external'
#<Encoding:IBM866>
```
I should mention probably that OS is Windows 7

**Comment by pocke** [comment]
Thank you. Maybe it is a platform issue or encoding issue. 
But I don't have a Windows machine, and my OS (Arch Linux) does not support IBM866 😿 

Maybe, you can fix this problem with adding `-E UTF-8` option to Ruby. It changes `Encoding.default_external` to UTF-8.
You can use `RUBYOPT` environment variable to specify the option. For example: `RUBYOPT='-E UTF-8'`. We can put the environment variable before rubocop command like `RUBYOPT='-E UTF-8' rubocop` in Linux, but I don't know how to specify an environment variable in Windows, sorry.

**Comment by yaxartes** [comment]
@pocke Thanks a lot! It's definately an encoding issue.
IBM866 (CP866) is the cyrillic version of CP437 developed in USSR 😄
 I set my `RUBYOPT` environment variable to `-E UTF-8`, and now `rubocop --auto-correct` works as it should.

**Comment by pocke** [comment]
No problem!

@bbatsov @Drenmi @jonas054 @koic Should we fix the issue? In other words, should RuboCop work in an encoding other than UTF-8 and ASCII?
I'm not sure we should fix it as I'm not familiar with encoding.

**Comment by jonas054** [comment]
We should at least try. If it's too difficult we can skip it.

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.

**Comment by ocarreterom** [comment]
Hi! I'm having this issue using vscode editor, but works fine in the terminal.

```sh
# system terminal
> echo $LANG # es_ES.UTF-8
> ruby -e 'p Encoding.default_external' # #<Encoding:UTF-8>

# vscode integrated terminal
> echo $LANG # en_US.UTF-8
> ruby -e 'p Encoding.default_external' # #<Encoding:UTF-8>
```

Adding `export RUBYOPT='-E UTF-8'` to `~/.zshrc` fix this problem, but why?
I am missing some config?

System info: macOS 13.1 (Apple M1)
